modules/web/js/ballerina/components/transform-statement-decorator.jsx

```diff
@@ -769,39 +756,63 @@ class TransformStatementDecorator extends React.Component {
     }
 
     setSource(currentSelection, predefinedStructs) {
-        let sourceSelection = _.find(predefinedStructs, { name: currentSelection });
-        if (_.isUndefined(sourceSelection)) {
-            alerts.error(`Mapping source "${  currentSelection  }" cannot be found`);
+        var sourceSelection =  _.find(predefinedStructs, { name:currentSelection});
+        if (_.isUndefined(sourceSelection)){
+            alerts.error('Mapping source "' + currentSelection + '" cannot be found');
             return false;
         }
+
+        var removeFunc = function(id) {
+            self.mapper.removeType(id);
+            _.remove(self.props.model.getInput(),  function(currentObject) {
+                                                                return currentObject.getVariableName() === id;
+                                                           });
+            self.props.model.setInput(self.props.model.getInput());
+            var currentSelectionObj =  _.find(self.predefinedStructs, { name:id});
+            currentSelectionObj.added = false;
+        }
+
         if (!sourceSelection.added) {
             if (sourceSelection.type == 'struct') {
-                self.mapper.addSourceType(sourceSelection);
+                self.mapper.addSourceType(sourceSelection, removeFunc);
             } else {
-                self.mapper.addVariable(sourceSelection, 'source');
+                self.mapper.addVariable(sourceSelection, 'source', removeFunc);
             }
             sourceSelection.added = true;
             return true;
-        }
-        return false;
+        } 
+            return false;
+        
     }
 
     setTarget(currentSelection, predefinedStructs) {
-        let targetSelection = _.find(predefinedStructs, { name: currentSelection });
-        if (_.isUndefined(targetSelection)) {
-            alerts.error(`Mapping target "${  currentSelection  }" cannot be found`);
+        var targetSelection = _.find(predefinedStructs, { name: currentSelection});
+        if (_.isUndefined(targetSelection)){
+            alerts.error('Mapping target "' + currentSelection + '" cannot be found');
             return false;
         }
+
+        var removeFunc = function(id) {
+            self.mapper.removeType(id);
+            _.remove(self.props.model.getOutput(), function(currentObject) {
+                                                                return currentObject.getVariableName() === id;
+                                                            })
```

Comment:
Is there an indentation issue here ? , I think we can use arrow function here as well.